File: sprint2_Toothless world/BE/be_sprint2/src/main/java/com/example/be_sprint2/service/ProductService.java
package com.example.be_sprint2.service;

import com.example.be_sprint2.dto.ProductDto;
import com.example.be_sprint2.model.product.Product;
import com.example.be_sprint2.repository.IProductRepository;
import com.example.be_sprint2.service.impl.IProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class ProductService implements IProductService {
    @Autowired
    private IProductRepository repository;


    @Override
    public Optional<Product> findProductById(Integer id) {
        return repository.findById(id);
    }



    @Override
    public Page<ProductDto> findAll(Pageable pageable, String name, String sizeName, String categoryName,Integer minPrice,Integer maxPrice) {
        return repository.findAllProDuct(pageable,"%"+name+"%","%"+sizeName+"%","%"+categoryName+"%",minPrice,maxPrice);
    }
}

File: sprint2_Toothless world/FE/fe-sprint2/src/service/ProductService.js
import axios from "axios";


export const findAll = async (currentPage,productName,sizeName,categoryName,minPrice,maxPrice) => {
    try {
        console.log(currentPage)
        let res = await axios.get(
            `http://localhost:8080/api/product?page=${currentPage}&name=${productName}&sizeName=${sizeName}&categoryName=${categoryName}&minPrice=${minPrice}&maxPrice=${maxPrice}`)
        return res
    } catch (e) {
        console.log(e)
    }

}

export const getProductDetails = async (id) => {
    try {
        const res = await axios.get(`http://localhost:8080/api/product/details/${id}`);
        return res;
    } catch (e) {
        alert("Hiển thị loại lỗi!");
    }
}

export const getAllCategory = async () => {
    try {
        let res = await axios.get(`http://localhost:8080/api/product/category`)
        return res
    } catch (e) {
        console.log(e)
    }

}
export const getAllSize = async (currentPage,sizeName,categoryName,productName) => {
    try {
        let res = await axios.get(`http://localhost:8080/api/product/size`)
        return res
    } catch (e) {
        console.log(e)
    }

}